bundles/org.eclipse.equinox.common/src/org/eclipse/core/text/StringMatcher.java

```diff
@@ -141,19 +169,79 @@ public String toString() {
 	 * @throws IllegalArgumentException if {@code pattern == null}
 	 */
 	public StringMatcher(String pattern, boolean ignoreCase, boolean ignoreWildCards) {
+		this(pattern, ignoreCase, ignoreWildCards, true);
+	}
+
+	/**
+	 * StringMatcher constructor takes in a String object that is a simple pattern.
+	 * The pattern may contain '*' for 0 and many characters and '?' for exactly one
+	 * character.
+	 * <p>
+	 * Literal '*' and '?' characters must be escaped in the pattern e.g., "\*"
+	 * means literal "*", etc.
+	 * </p>
+	 * <p>
+	 * The escape character '\' is an escape only if followed by '*', '?', or '\'.
+	 * All other occurrences are taken literally.
+	 * </p>
+	 * <p>
+	 * If invoking the StringMatcher with string literals in Java, don't forget
+	 * escape characters are represented by "\\".
+	 * </p>
+	 * <p>
+	 * If {@code ignoreWords} is true, this {@code StringMatcher} matches a pattern
+	 * that may contain the wildcards '?' or '*' against a text. However, the
+	 * matching is not only done on the full text, but also on individual words from
+	 * the text, and if the pattern contains whitespace, the pattern is split into
+	 * sub-patterns and those are matched, too.
+	 * </p>
+	 * <p>
+	 * The precise rules are:
+	 * </p>
+	 * <ul>
+	 * <li>Leading and trailing whitespace in the pattern is ignored.</li>
+	 * <li>If the full pattern matches the full text, the match succeeds.</li>
+	 * <li>If the full pattern matches a single word of the text, the match
+	 * succeeds.</li>
+	 * <li>If all sub-patterns match a prefix of the whole text or any prefix of any
+	 * word, the match succeeds.</li>
+	 * <li>Otherwise, the match fails.</li>
+	 * </ul>
+	 * <p>
+	 * An empty pattern matches only an empty text, unless {@link #usePrefixMatch()}
+	 * is used, in which case it always matches.
+	 * </p>
+	 *
+	 * @param pattern         the pattern to match text against, must not be
+	 *                        {@code null}
+	 * @param ignoreCase      if true, case is ignored
+	 * @param ignoreWildCards if true, wild cards and their escape sequences are
+	 *                        ignored (everything is taken literally).
+	 * @param ignoreWords     if true, only matches against the whole text but not
+	 *                        individual words
+	 * @throws IllegalArgumentException if {@code pattern == null}
+	 * @since 3.20
+	 */
+	public StringMatcher(String pattern, boolean ignoreCase, boolean ignoreWildCards, boolean ignoreWords) {
 		if (pattern == null) {
 			throw new IllegalArgumentException();
 		}
 		fIgnoreCase = ignoreCase;
 		fIgnoreWildCards = ignoreWildCards;
-		fPattern = pattern;
-		fLength = pattern.length();
+		fIgnoreWords = ignoreWords;
+		fPattern = pattern.trim();
+		fLength = fPattern.length();
```

Comment:
I wonder if there is a better way to handle this here... Originally, the Platform was using a copy of this StringMatcher, where it would explicitly trim the input pattern. See https://bugs.eclipse.org/bugs/show_bug.cgi?id=570390

Reply:
It is unclear to me on if this could break existing uses of `StringMatcher` where `ignoreWords` would be `true`.

I also doubt you intended to `trim` here and also in `splitPattern` (https://github.com/eclipse-equinox/equinox/pull/720/files#diff-ba6befa586c50334dce15c73bd18a046e4c1194bd272fd52f461ceacf7b1ee7aR479)

Reply:
> It is unclear to me on if this could break existing uses of StringMatcher where ignoreWords would be true.

That's what I wasn't sure about, either. I've updated it so that the string is only trimmed if `ignoredWords` is `false`, i.e. when the new functionality is used.

> I also doubt you intended to trim here and also in splitPattern (https://github.com/eclipse-equinox/equinox/pull/720/files#diff-ba6befa586c50334dce15c73bd18a046e4c1194bd272fd52f461ceacf7b1ee7aR479)

True, that was a left-over from the original implementation, which I've now also updated.